src/kubernetes-ingress-controller/references/custom-resources.md

```diff
@@ -1,458 +1,429 @@
-
-If `host` is not specified, then port-based TCP routing is performed. Kong
-doesn't care about the content of TCP stream in this case.
-
-If `host` is specified, then Kong expects the TCP stream to be TLS-encrypted
-and Kong will terminate the TLS session based on the SNI.
-Also note that, the port in this case should be configured with `ssl` parameter
-in Kong.
-
-## UDPIngress
-
-The `UDPIngress` API makes it possible to route traffic to your [UDP][udp] services
-using Kong (e.g. DNS, Game Servers, e.t.c.).
-
-```yaml
-apiVersion: configuration.konghq.com/v1beta1
-kind: UDPIngress
-metadata:
-  name: <object name>
-  namespace: <object namespace>
-  annotations:
-    kubernetes.io/ingress.class: <controller ingress class, "kong" by default>
-spec:
-  rules:
-  - port: <port on which to expose this service, required>
-    backend:
-      serviceName: <name of the kubernetes service, required>
-      servicePort: <port number to forward on the service, required>
-```
-
-For each rule provided in the spec the Kong proxy environment must be updated to
-listen to UDP on that port as well.
-
-[udp]:https://datatracker.ietf.org/doc/html/rfc768
-
-## KongConsumer
-
-This custom resource configures a consumer in Kong:
-
-The following snippet shows the field available in the resource:
-
-```yaml
-apiVersion: configuration.konghq.com/v1
-kind: KongConsumer
-metadata:
-  name: <object name>
-  namespace: <object namespace>
-  annotations:
-    kubernetes.io/ingress.class: <controller ingress class, "kong" by default>
-username: <user name>
-custom_id: <custom ID>
-```
-
-An example:
-
-```yaml
-apiVersion: configuration.konghq.com/v1
-kind: KongConsumer
-metadata:
-  name: consumer-team-x
-  annotations:
-    kubernetes.io/ingress.class: kong
-username: team-X
-credentials:
-  - secretRef1
-  - secretRef2
-```
-
-When this resource is created, a corresponding consumer entity will be
-created in Kong.
-
-Consumers' `username` and `custom_id` values must be unique across the Kong
-cluster. While KongConsumers exist in a specific Kubernetes namespace,
-KongConsumers from all namespaces are combined into a single Kong
-configuration, and no KongConsumers with the same `kubernetes.io/ingress.class`
-may share the same `username` or `custom_id` value.
-
-For help configuring credentials for the `KongConsumer` Please refer to the [using the Kong Consumer and Credential resource](/kubernetes-ingress-controller/{{page.kong_version}}/guides/using-consumer-credential-resource) guide.
-
-[k8s-crd]: https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/
-[kong-consumer]: /gateway/latest/admin-api/#consumer-object
-[kong-plugin]: /gateway/latest/admin-api/#plugin-object
-[kong-upstream]: /gateway/latest/admin-api/#upstream-object
-[kong-service]: /gateway/latest/admin-api/#service-object
-[kong-route]: /gateway/latest/admin-api/#route-object
+[This document is generated by KIC's 'generate.docs' make target, DO NOT EDIT]: #
```

Comment:
Any reason this is a link definition instead of a comment? I guess it doesn't render, but I am confused by the current use of Markdown here.

Reply:
I used this way to include a comment following the top-voted answer to "[Comments in Markdown](https://stackoverflow.com/questions/4823468/comments-in-markdown)" on SO. I missed the part telling it's the best solution if you want to make the comment not included in the resulting HTML. For us, it shouldn't be a problem even if the comment ends up in the resulting HTML. I will change it to a regular `<!-- -->` style comment.

Reply:
https://github.com/Kong/kubernetes-ingress-controller/pull/3210